tests/templates/test_subroutines/test_qsvt.py

```diff
@@ -0,0 +1,173 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for the QSVT template and qsvt wrapper function.
+"""
+import pytest
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.templates.subroutines.qsvt import *
+
+
+class TestQSVT:
+    @pytest.mark.parametrize(
+        ("U_A", "lst_projectors", "wires", "expected_output"),
+        [
+            (
+                qml.BlockEncode([[0.1, 0.2], [0.3, 0.4]], wires=[0, 1]),
+                [qml.PCPhase(0.5, dim=2, wires=[0, 1]), qml.PCPhase(0.5, dim=2, wires=[0, 1])],
+                [0, 1],
+                0.06587008470151662,
+            ),
+            (
+                qml.BlockEncode([[0.3, 0.1], [0.2, 0.9]], wires=[0, 1]),
+                [qml.PCPhase(0.5, dim=2, wires=[0, 1]), qml.PCPhase(0.3, dim=2, wires=[0, 1])],
+                [0, 1],
+                0.24662762785797077,
+            ),
+        ],
+    )
+    def test_output(self, U_A, lst_projectors, wires, expected_output):
+        dev = qml.device("default.qubit", wires=len(wires))
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.QSVT(U_A, lst_projectors, wires)
+            return qml.expval(qml.PauliY(wires=0))
+
+        assert np.isclose(circuit(), expected_output)
+
+    @pytest.mark.parametrize(
+        ("U_A", "lst_projectors", "wires", "results"),
+        [
+            (
+                qml.BlockEncode(0.1, wires=0),
+                [qml.PCPhase(0.2, dim=1, wires=0), qml.PCPhase(0.3, dim=1, wires=0)],
+                0,
+                [
+                    qml.BlockEncode(np.array([[0.1]]), wires=[0]),
+                    qml.PCPhase(0.2, dim=2, wires=[0]),
+                    qml.adjoint(qml.BlockEncode(np.array([[0.1]]), wires=[0])),
+                    qml.PCPhase(0.3, dim=2, wires=[0]),
+                ],
+            ),
+            (
+                qml.PauliZ(wires=0),
+                [qml.RZ(0.1, wires=0), qml.RY(0.2, wires=0), qml.RZ(0.3, wires=1)],
+                [0, 1],
+                [
+                    qml.PauliZ(wires=[0]),
+                    qml.RZ(0.1, wires=[0]),
+                    qml.adjoint(qml.PauliZ(wires=[0])),
+                    qml.RY(0.2, wires=[0]),
+                    qml.PauliZ(wires=[0]),
+                    qml.RZ(0.3, wires=[1]),
+                ],
+            ),
+        ],
+    )
+    def test_queuing_ops(self, U_A, lst_projectors, wires, results):
+        with qml.tape.QuantumTape() as tape:
+            qml.QSVT(U_A, lst_projectors, wires)
+
+        for idx, val in enumerate(tape.expand().operations):
+            assert val.name == results[idx].name
+            assert val.parameters == results[idx].parameters
+
+    def test_queuing_ops_defined_in_circuit(self):
```

Comment:
Not sure I understand what this function tests.

Reply:
This tests that the block encoding and projector-controlled phase shift operations are queued in the right order, without any extra queuings.